src/Shared/Helpers/OutputBuilder/SarifOutputBuilder.cs

```diff
@@ -0,0 +1,145 @@
+﻿using Microsoft.CodeAnalysis.Sarif;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using SarifResult = Microsoft.CodeAnalysis.Sarif.Result;
+
+namespace Microsoft.CST.OpenSource.Shared
+{
+    public class SarifOutputBuilder : IOutputBuilder
+    {
+        /// <summary>
+        ///     Class logger
+        /// </summary>
+        protected static readonly NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();
+
+        // cache variables to avoid reflection
+        private static readonly string AssemblyName = Assembly.GetEntryAssembly()?.GetName().Name ?? string.Empty;
+
+        private static readonly string Company = Assembly.GetEntryAssembly()?.GetCustomAttribute<AssemblyCompanyAttribute>()?.Company ?? string.Empty;
+        private static readonly string Version = Assembly.GetEntryAssembly()?.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version.ToString() ?? string.Empty;
+        private readonly SarifVersion currentSarifVersion = SarifVersion.Current;
+        private List<Result> sarifResults = new List<Result>();
+        // default = text
+
+        public SarifOutputBuilder(SarifVersion version)
+        {
+            currentSarifVersion = version;
+        }
+
+        /// <summary>
+        ///     Build a SARIF Result.Location object for the purl package
+        /// </summary>
+        /// <param name="purl"> </param>
+        /// <returns> Location list with single location object </returns>
+        /// <returns> Location list with single location object </returns>
+        public static List<Location> BuildPurlLocation(PackageURL purl)
+        {
+            BaseProjectManager? projectManager = ProjectManagerFactory.CreateProjectManager(purl, null);
+            if (projectManager == null)
+            {
+                Logger?.Error("Cannot determine the package type");
```

Comment:
Should Logger be nullable anywhere? I didn't know why nullable Loggers were added so I left it there...I can remove all the nullable Loggers

Reply:
Nullable loggers were added in the first pass of nullability because the way they were being referenced allowed them to be null in some cases, and my initial implementation of nullability was to denote what is.  That is fixed now with just declaring a protected logger in each class so logger should no longer be null anywhere.

Reply:
changed all Logger?. to Logger.